app-modules/report/database/factories/ReportFactory.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2022-2023, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Report\Database\Factories;
+
+use App\Models\User;
+use AdvisingApp\Report\Models\Report;
+use AdvisingApp\Report\Enums\ReportModel;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+/**
+ * @extends Factory<Report>
+ */
+class ReportFactory extends Factory
+{
+    /**
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'name' => fake()->words(asText: true),
+            'model' => fake()->randomElement(ReportModel::cases()),
+        ];
+    }
+
+    public function configure(): ReportFactory|Factory
+    {
+        return $this->afterMaking(function (Report $report) {
+            $report->user()->associate(User::inRandomOrder()->first() ?? User::factory()->create());
+        });
+    }
```

Comment:
Is there a reason to be doing this here rather than in `user_id` in the definition?

Reply:
It's the same in the CaseloadFactory, do you want to change it?

I think the reason is so that it uses existing users, instead of creating a new user each time

Reply:
Yeah, I would prefer we change it to be in the definition.

We can do the existing users check inline within the definition. There are other factories in the application where we do this.

I would like to move away from this after making part where we can and stick to just the definition.

Reply:
done